### PR TITLE
feat(chat): Message display several images

### LINKED ISSUE #4905: sending two images is allowed but only one image is actually sent
# Bug Report

## Steps to reproduce

1. join a community channel
2. drag and drop an image (but do not send yet)
3. drag and drop another image
4. send

Screen Shot 2022-02-25 at 3 55 17 PM

#### Expected behavior

If the spec allows it, then both images should be displayed. if the spec does not allow it, then only one image should be allowed to be attached

#### Actual behavior

Only one image is attached

Screen Shot 2022-02-25 at 3 55 22 PM

### PR DESCRIPTION
**Pull request by MishkaRogachev**
Close #4905

### What does the PR do

Change rules how we get response on sent messages

### Affected areas

Chat

### Screenshot of functionality

https://user-images.githubusercontent.com/2522130/177334693-ebf4bde8-a010-4107-b934-6e8ab802e2e3.mov

### CODE REVIEW

**Comment by MishkaRogachev** [comment]
@Khushboo-dev-cpp @saledjenic cane you please take a look, i've did not found any documentation why `processMessageUpdateAfterSend` was done that way.

**Review by jrainville — COMMENTED** [review]
Code looks good. Have you tested if sending a reply or receiving a reply does something weird, since you changed the code for handling replies. I think it should be ok, but better safe than sorry.

**Comment by MishkaRogachev** [comment]
> Code looks good. Have you tested if sending a reply or receiving a reply does something weird, since you changed the code for handling replies. I think it should be ok, but better safe than sorry.

Yes, i didn't notice something weird. But I want to wait @saledjenic and @Khushboo-dev-cpp reviews.

**Review by elina2015 — APPROVED** [review]
Tested and approved

**Review by jrainville — APPROVED** [review]

File: src/app_service/service/chat/service.nim [hunk 1]
```diff
@@ -204,24 +203,14 @@ QtObject:
...
 
-  proc processMessageUpdateAfterSend*(self: Service, response: RpcResponse[JsonNode]): (seq[ChatDto], seq[MessageDto])  =
+  proc processMessageUpdateAfterSend*(self: Service, response: RpcResponse[JsonNode]): (seq[ChatDto], seq[MessageDto]) =
     result = self.parseChatResponse(response)
     var (chats, messages) = result
     if chats.len == 0 or messages.len == 0:
       error "no chats or messages in the parsed response"
       return
 
-    # The reason why we are sending all the messages with responseTo filled in is because
```
Comment (Khushboo-dev-cpp) [inline src/app_service/service/chat/service.nim:214]: hey @MishkaRogachev this as far as I remember was done cause the reply wasnt coming filled correctly from status_go and we had to manually update it in some cases, i do not remember the exactly use case that this didnt work in though

Comment (MishkaRogachev) [reply src/app_service/service/chat/service.nim:214]: Thanks) I've tested with replies and it seems to be working well. I will wait for @saledjenic probably he remember some use case

Comment (saledjenic) [reply src/app_service/service/chat/service.nim:214]: @MishkaRogachev I remember that I added that in the commit `58f10babba868fa2d2e99a779f8358b8d0ed3eac` for the old status-lib repo, which is not in use any more. And it was almost a year ago, how that ended up in the refactored code I am not sure, as I see it was added in the commit `a9c968b984d15f2c74e42689007d0fc88fb1086a `. All in all that is most outdated code, so you are free to remove it and if that causes new issue you can log it and we can fix it according to the code we're currently maintaining.

**Comment by MishkaRogachev** [comment]
@saledjenic @iurimatias @jrainville @Khushboo-dev-cpp I've found some problems but they also affects master before my applying my changes: https://github.com/status-im/status-desktop/issues/6407

**Pull request merged**